### ISSUE TITLE
Not able to get the script from Admin Account, I dont want to use it for a single project.

### ISSUE THREAD

**Issue by shivaniverma6991** [doitintl/SafeScrub#1]

**Comment by JoshuaFox** [comment]
@shivaniverma6991  What is an Admin Account? Do you mean an account that has permissions to  access multiple projects? I suggest just iterating over the projects as follows. 

Does that give you what you need?


```
projects=( proj1 proj2)
for project in "${projects[@]}"
do
./generate-deletion-script.sh \
  -p  ${project} \
  -k admin-credentials.json \
  >> deletion-script.sh
done

chmod a+x deletion-script.sh
```

**Comment by shivaniverma6991** [comment]
@JoshuaFox Yes, I mean I don't want to use Service Account that will be limited to one Project. One account that has access to everything, I want to use that one, where do I get the credentials.json for that account? Can I do it without credentials, followed by initial login?

**Comment by JoshuaFox** [comment]
@shivaniverma6991 Please clarify

 
> Yes, I mean I don't want to use Service Account that will be limited to one Project. 
> One account that has access to everything, I want to use that one,

Do you mean that 
1. You want to use a Service Account which has access to multiple projects? 
If so, you can assign a Role (like Project Viewer) to that Service Account from each of the projects.
2. Or that you do not want to use a Service Account, insteast using your user account (which has access to multiple projects)?
You can do that too. Safe Scrub uses a Service Account json credentials file to require explicit decision on the permissions used, but if you are willing to change that, then modify the source code to delete `  =gcloud -q auth activate-service-account --key-file "${key_file}" || exit 1` and your logged-in gcloud user account will be used.

**Comment by shivaniverma6991** [comment]
@JoshuaFox point 2 is exactly what I need, thanks I got it working, but I need more services also i.e BigQuery, Spanner, Cloud SQL, IAM, BigTable

**Comment by JoshuaFox** [comment]
> @JoshuaFox point 2 is exactly what I need, thanks I got it working

Good to know.
> but I need more services also i.e BigQuery, Spanner, Cloud SQL, IAM, BigTable
Cloud SQL is supported.

Feel free to submit a Pull Request for the others. I may get to these as well.

**Comment by shivaniverma6991** [comment]
@JoshuaFox Not able to submit pull request man, Let me know if you release others shortly and do you have idea if I can run this script from Cloud Functions?

**Comment by shivaniverma6991** [comment]
@JoshuaFox Does it delete load balancers?

**Comment by JoshuaFox** [comment]
Safe Scrub does list load balancers. (Safe Scrub  does not actual *delete* anything.)

Load balancers are managed in `gcloud` through their parts and each of these are listed.

```
backend-services
forwarding-rules
health-checks 
http-health-checks 
https-health-checks
target-pools
```

**Comment by shivaniverma6991** [comment]
@JoshuaFox I am able to create deletion script for all the resources but Can you please help with deletion of BigQuery?

**Comment by shivaniverma6991** [comment]
@JoshuaFox Can I restrict the IAM user to a single region?

**Comment by JoshuaFox** [comment]
I don't think that that is a feature of IAM in general (without any connection to this tool.)

**Comment by shivaniverma6991** [comment]
Can we customize such a role? like there in AWS?

**Comment by JoshuaFox** [comment]
That is a general GCP question and not related specifically to Safe Scrub. However, I believe that the answer is  no. GCP has a more global design than AWS. Though resources may live ina. region, much more of the access and control is global  compared to AWS
